src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -86,27 +97,54 @@ public List<Section> getSections() {
         return Collections.unmodifiableList(sections);
     }
 
-    public Section getDeleteTargetSection(Long stationId) {
+    public void deleteSection(Station station) {
         if (sections.size() <= 1) {
             throw new CannotDeleteSectionException(ErrorMessage.CANNOT_DELETE_SECTION);
         }
 
-        if (!isLastStation(stationId)) {
-            throw new CannotDeleteSectionException(ErrorMessage.CANNOT_DELETE_SECTION);
+        List<Section> targetSections = sections.stream().filter(section -> section.containStation(station))
+                .collect(Collectors.toList());
+
+        if (targetSections.isEmpty()) {
+            throw new CannotDeleteSectionException(ErrorMessage.NO_STATION_EXIST);
         }
 
-        return getLastSection();
+        // 노선에 존재하는 상행역, 하행역을 삭제하는 경우
+        if (isFirstStationOrLastStation(targetSections)) {
+            Section section = targetSections.get(0);
+            deleteExistSection(section);
+            return;
+        }
+
+        // 노선 중간에 존재하는 역을 삭제하는 경우
+        deleteMiddleStation(targetSections);
```

Comment:
```suggestion
        if (isFirstStationOrLastStation(targetSections)) {
            Section section = targetSections.get(0);
            deleteExistSection(section);
        } else {
             // 노선 중간에 존재하는 역을 삭제하는 경우
            deleteMiddleStation(targetSections);
        }
```

스타일이긴 한데요~ early return 은 검증에 대해 저는 자주 사용하고, 이처럼 서비스 로직이 각각 실행 될 때는 if/else로 묶어주고는 있습니다.